src/chef_ci_status.rb

```diff
@@ -194,19 +335,152 @@ def print_pr_or_issue_stats(stats, item)
   puts "    Avg Time to Close #{item_plural}: #{avg_time_str}"
 end
 
-options = {
-  org: 'chef',
-  repo: 'chef',
-  branches: ['main'],
-  days: 30,
-  verbose: false,
-  mode: ['all'],
-}
+# Define defaults from config or use hardcoded values
+options = if defined?(Settings)
+            default_mode = if Settings.default_mode.is_a?(Array)
+                             Settings.default_mode
+                           else
+                             [Settings.default_mode]
+                           end
+            # Get CI timeout from config with fallback to default
+            # Get timeout with fallback
+            has_timeout = Settings.respond_to?(:ci_timeout)
+            ci_timeout = has_timeout ? Settings.ci_timeout : 180
+            {
+              org: Settings.default_org,
+              repo: Settings.default_repo,
+              branches: Settings.default_branches,
+              days: Settings.default_days,
+              verbose: false,
+              mode: default_mode,
+              ci_timeout:, # From config or default: 3 minutes
+            }
+          else
+            {
+              org: 'chef',
+              repo: 'chef',
+              branches: ['main'],
+              days: 30,
+              verbose: false,
+              mode: ['all'],
+              ci_timeout: 180, # Default: 3 minutes
+            }
+          end
 
 valid_modes = %w{ci pr issue all}
 OptionParser.new do |opts|
   opts.banner = 'Usage: chef_ci_status.rb [options]'
 
+  opts.on(
+    '--config CONFIG_FILE',
+    'Path to custom configuration file',
+  ) do |v|
+    if File.exist?(v)
+      begin
+        # Load and parse the YAML file
+        custom_config = begin
+          YAML.load_file(v)
+                        rescue Psych::SyntaxError => e
+                          raise "Invalid YAML syntax in #{v}: #{e.message}"
+        end
+
+        # Verify it's a hash/dictionary
+        unless custom_config.is_a?(Hash)
+          raise 'Configuration file must contain a YAML dictionary/hash, ' \
+                "not a #{custom_config.class}"
+        end
+
+        # Basic validation of required configuration keys
+        required_keys = %w{default_org default_repo default_branches
+default_days}
+        missing_keys = required_keys.select { |key| !custom_config.key?(key) }
```

Comment:
we have validation code, why repeat it all here? lets load a validator, and validate. Then we can make a `chef_ci_status.rb --mode validate-config` and there's not 3 validators.